### DISCUSSION TITLE
rework of expand shortcode.. (can submit PR if desired)

### DISCUSSION THREAD

**Discussion by wolfspyre** [thegeeklab/hugo-geekdoc#707] [Ideas]
hiya!!

I modified the expand shortcode a bit, and pushed a branch to my fork for you to assess if you'd like to integrate it.

https://github.com/wolfspyre/hugo-geekdoc/tree/expand

I wanted to get footnotes to work within expands.... 

**cue yak shaving adventure**

[This Page by Nelis Oostens](https://oostens.me/posts/hugo-shortcodes-with-markdown-gotchas/) was really enlightening and admittedly the basis for the change.

I also added a param to have the expander open by default.
and one to set the prefix id as opposed to the programatically generated sha... 
I'm not especially attached to that; but figured It wouldn't hurt.

LMK if you'd like me to sub as a PR, or if you want changes, or are happy with the existing functionality as-is.

Thanks for making geekdoc! Just trying to give back.

**Comment by xoxys** [comment]
Hi, please see my answer in https://github.com/thegeeklab/hugo-geekdoc/issues/436#issuecomment-1152265018 why this was not added to the theme.
